app/src/parser_impl.c

```diff
@@ -1256,6 +1367,17 @@ static parser_error_t _readTxApplication(parser_context_t *c, parser_tx_t *v)
         return parser_program_fields_too_long;
     }
 
+    // Display reject_version and access_list at the end (matching enum order)
+    if (application->reject_version > 0) {
+        DISPLAY_ITEM(IDX_REJECT_VERSION, 1, tx_num_items)
+    }
+
+    // Store the display offset before adding access_list items
+    application->access_list_display_offset = tx_num_items;
+    if (application->num_access_list_element > 0) {
```

Comment:
I'd suggest we stick to using something like:
```suggestion
    if (_findKey(c, KEY_APP_ACCESS_LIST) == parser_ok) {
```
I get that it's not as efficient as checking how many elements you've already found, but like above, I think it's best we stick to existing conventions rather than mix-match. Also, this advances the buffer pos so it's exactly where you want it to be, so it removes the need for the `access_list_display_offset`.